includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -4434,6 +4435,28 @@ private function upe_needs_redirection( $payment_methods ) {
 		return 1 === count( $payment_methods ) && 'card' !== $payment_methods[0];
 	}
 
+
+	/**
+	 * Disable WooPay.
+	 *
+	 * This function disables WooPay by updating the option to 'no'
+	 * once the 'disable_woopay' filter is called.
+	 *
+	 * @return boolean False, indicating WooPay is disabled.
+	 */
+	public function disable_woopay() {
+		// Filter: disable_woopay.
+		$disable = apply_filters( 'disable_woopay', true );
```

Comment:
This disables WooPay even when nothing is hooking to this filter. Changing the default value to `false` makes it work as expected